src/NServiceBus.Transport.RabbitMQ.TransportTests/ConfigureRabbitMQTransportInfrastructure.cs

```diff
@@ -55,7 +46,38 @@ public Task Cleanup()
 
     void PurgeQueues()
     {
-        var connectionFactory = CreateConnectionFactory();
+        if (connectionStringBuilder == null)
```

Comment:
Something about this whole pattern doesn't sit right with me but I can't quite put my finger on it.

Anyway, I'll make a change to ensure that `Cleanup` only runs if `Initialize` succeeded.